### ISSUE TITLE
rule: scroll-no-scroll-padding

### ISSUE THREAD

**Issue by purupurupu** [purupurupu/css-noop-checker#133]
## Rule ID
`scroll-no-scroll-padding`

## What does this rule detect?
`scroll-padding` and its sub-properties only apply to scroll containers (elements with `overflow: scroll`, `auto`, or `hidden`). On elements with `overflow: visible`, they have no effect.

## Example: should warn
```html
<div style="overflow: visible; scroll-padding: 20px;">content</div>
```

## Example: should NOT warn
```html
<div style="overflow: auto; scroll-padding: 20px;">content</div>
```

## Related CSS Properties
`overflow`, `scroll-padding`, `scroll-padding-top`, `scroll-padding-right`, `scroll-padding-bottom`, `scroll-padding-left` (and logical equivalents)

## References
- [CSS Scroll Snap Level 1 §6 - scroll-padding](https://drafts.csswg.org/css-scroll-snap/#scroll-padding): applies to scroll containers
- [Firefox inactive-property-helper.js](https://searchfox.org/mozilla-central/source/devtools/server/actors/utils/inactive-property-helper.js) — detects this pattern
- Becoming more relevant with scroll-snap adoption
